esp-hal/src/dma/mod.rs

```diff
@@ -1739,8 +1775,17 @@ where
                     self.set_ext_mem_block_size(block_size.into());
                 }
             } else {
-                if is_slice_in_psram(self.buffer) {
-                    return Err(DmaError::UnsupportedMemoryRegion);
+                // we check here to insure that no descriptor points to PSRAM as
+                // that is only supported on esp32s3
+                let mut descriptor = Some(*preparation.start);
+                while let Some(current) = descriptor {
+                    if is_slice_in_psram(core::slice::from_raw_parts(current.buffer, current.len())) {
+                        return Err(DmaError::UnsupportedMemoryRegion);
```

Comment:
Oops, I just realised this gets in the way of what I was telling @bugadani in https://github.com/esp-rs/esp-hal/pull/2191#discussion_r1765588448 .

I'll deal with this in a separate PR I think. The only way to avoid this check would be to use the type system or panic.